server/datasets/tcga/ingest.py

```diff
@@ -25,392 +25,309 @@
-                                    tcgaBaseUrl, diseaseStudyCode, repositoryType,
-                                    dataProvider, dataType, 'slide_images')[0],
-                                dataProvider, diseaseStudyCode, dataType):
-
-                            for slideFileName, modifiedTime in listAutoIndex(
-                                    tcgaBaseUrl, diseaseStudyCode, repositoryType, dataProvider,
-                                    dataType, 'slide_images', batchDirectory)[1]:
-                                if not slideFileName.endswith('.svs'):
-                                    continue
-
-                                slideFileUrl = '/'.join([
-                                    tcgaBaseUrl, diseaseStudyCode, repositoryType,
-                                    dataProvider, dataType, 'slide_images',
-                                    batchDirectory, slideFileName])
-
-                                slideBarcodeMatch = re.match(
-                                    r'^TCGA-'
-                                    r'(?P<TSS>\w{2})-'  # Tissue source site ID
-                                    r'(?P<Participant>\w{4})-'  # Study participant
-                                    r'(?P<Sample>\d{2})'  # Sample type
-                                    r'(?P<Vial>[A-Z])-'  # Order of sample in a sequence of samples
-                                    r'(?P<Portion>\d{2})-'  # Order of portion in a sequence of 100 - 120 mg sample portions
-                                    r'(?P<SlideLocation>TS|MS|BS|DX)'
-                                    r'(?P<SlidePortion>[0-9A-Z]?)\.'
-                                    # TODO: UUID should always be uppercase?
-                                    r'(?P<UUID>[0-9a-zA-Z]{8}-[0-9a-zA-Z]{4}-[0-9a-zA-Z]{4}-[0-9a-zA-Z]{4}-[0-9a-zA-Z]{12})\.'
-                                    r'svs$',
-                                    slideFileName
-                                )
-                                if not slideBarcodeMatch:
-                                    log('Could not parse slide barcode: "%s"' % slideFileUrl)
-                                    continue
-                                    raise MetadataParseException('Could not parse slide barcode: "%s"' % slideFileUrl)
-                                slideBarcode = slideBarcodeMatch.groupdict()
-
-                                if limit and ingestCount >= limit:
-                                    global ingestUser
-                                    ingestUser = None
-                                    collectionCache.clear()
-                                    folderCache.clear()
-                                    return
-                                if progress is not None:
-                                    progress.update(
-                                        current=ingestCount,
-                                        total=limit if limit else None,
```

Comment:
@manthey FYI, this was a bug. The `total` parameter must be an `int` or `float`.